src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -956,3 +943,98 @@ def assign_last_name_ids(self, pop: pd.DataFrame) -> pd.Series:
         )
 
         return pop["last_name_id"]
+
+    def update_reference_person_and_relationships(
+        self, population: pd.DataFrame
+    ) -> pd.DataFrame:
+        # Takes population state table and assigns reference person in households that do not have one after a time step
+        # After a new reference person has been assigned, biological children and other household members have their
+        #  relationships updated.
+        # Find standard households that do not have a reference person
+        household_ids_with_reference_person = population.loc[
+            population["relation_to_household_head"] == "Reference person", "household_id"
+        ]
+        standard_household_ids = population.loc[
+            population["housing_type"] == "Standard", "household_id"
+        ].unique()
+        household_ids_without_reference_person = set(standard_household_ids) - set(
+            household_ids_with_reference_person
+        )
+        households_to_update_idx = population.index[
+            population["household_id"].isin(household_ids_without_reference_person)
+        ]
+
+        # Find oldest member in each household and make them new reference person
+        # This is a series with household_id as the index and the new reference person as the value
+        new_reference_persons = (
+            population.loc[households_to_update_idx].groupby(["household_id"])["age"].idxmax()
+        )
+        # Preserve old relationship of new reference person before assigning them as new reference persons
+        new_reference_persons_relationship_to_old_reference_person = population.loc[
+            new_reference_persons, ["household_id", "relation_to_household_head"]
+        ]
+        population.loc[
+            new_reference_persons, "relation_to_household_head"
+        ] = "Reference person"
+
+        # Update simulants born in simulation as biological children to their mother if mother is new reference person
+        biological_children_idx = population.index[
+            (population["date_of_birth"] > self.start_time)
+            & (
```

Comment:
Consider storing these in intermediate variables so black doesn't make it illegible.

We should revisit our black settings to try to avoid things like this.